dds/DdsDcpsTypeSupportExt.idl

```diff
@@ -39,6 +39,11 @@ module OpenDDS {
   #endif
 
       boolean has_dcps_key();
+
+      ::DDS::ReturnCode_t unregister_type(
+          in ::DDS::DomainParticipant domain,
+          in string type_name);
```

Comment:
Is there a need for the type_name parameter here?  Couldn't the receiving object just be unregistered from the domain participant?  It already knows its own type_name.